From 55b9ae42a6cd1553a500852dfefeeff9af5d8b7f Mon Sep 17 00:00:00 2001
From: Cezary Rojewski <cezary.rojewski@intel.com>
Date: Mon, 8 Oct 2018 12:47:45 +0200
Subject: [PATCH 253/292] ASoC: Intel: Skylake: Do not use fw_ops handlers
 explicitly.

Driver should refer to specific handler via ops instead of using said
handler explicitly.

Change-Id: I8923c212504de18015e187dc7d47662bc1843efa
Signed-off-by: Cezary Rojewski <cezary.rojewski@intel.com>
Reviewed-on:
Reviewed-by: Slawinski, AmadeuszX <amadeuszx.slawinski@intel.com>
Tested-by: Slawinski, AmadeuszX <amadeuszx.slawinski@intel.com>
---
 sound/soc/intel/skylake/bxt-sst.c | 4 ++--
 sound/soc/intel/skylake/cnl-sst.c | 2 +-
 sound/soc/intel/skylake/skl-sst.c | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/skylake/bxt-sst.c b/sound/soc/intel/skylake/bxt-sst.c
index e5d7cb45a4e2..4b0649fa94b0 100644
--- a/sound/soc/intel/skylake/bxt-sst.c
+++ b/sound/soc/intel/skylake/bxt-sst.c
@@ -469,14 +469,14 @@ static int bxt_set_dsp_D0(struct sst_dsp *ctx, unsigned int core_id)
 
 	if (skl->fw_loaded == false) {
 		skl->boot_complete = false;
-		ret = bxt_load_base_firmware(ctx);
+		ret = ctx->fw_ops.load_fw(ctx);
 		if (ret < 0) {
 			dev_err(ctx->dev, "reload fw failed: %d\n", ret);
 			return ret;
 		}
 
 		if (skl->lib_count > 1) {
-			ret = bxt_load_library(ctx, skl->lib_info,
+			ret = ctx->fw_ops.load_library(ctx, skl->lib_info,
 						skl->lib_count);
 			if (ret < 0) {
 				dev_err(ctx->dev, "reload libs failed: %d\n", ret);
diff --git a/sound/soc/intel/skylake/cnl-sst.c b/sound/soc/intel/skylake/cnl-sst.c
index efe97e78a50e..55a44599a21c 100644
--- a/sound/soc/intel/skylake/cnl-sst.c
+++ b/sound/soc/intel/skylake/cnl-sst.c
@@ -284,7 +284,7 @@ static int cnl_set_dsp_D0(struct sst_dsp *ctx, unsigned int core_id)
 
 	if (!cnl->fw_loaded) {
 		cnl->boot_complete = false;
-		ret = cnl_load_base_firmware(ctx);
+		ret = ctx->fw_ops.load_fw(ctx);
 		if (ret < 0) {
 			dev_err(ctx->dev, "fw reload failed: %d\n", ret);
 			return ret;
diff --git a/sound/soc/intel/skylake/skl-sst.c b/sound/soc/intel/skylake/skl-sst.c
index 5951bbdf1f1a..85f8227cf2f9 100644
--- a/sound/soc/intel/skylake/skl-sst.c
+++ b/sound/soc/intel/skylake/skl-sst.c
@@ -174,7 +174,7 @@ static int skl_set_dsp_D0(struct sst_dsp *ctx, unsigned int core_id)
 
 	/* If core0 is being turned on, we need to load the FW */
 	if (core_id == SKL_DSP_CORE0_ID) {
-		ret = skl_load_base_firmware(ctx);
+		ret = ctx->fw_ops.load_fw(ctx);
 		if (ret < 0) {
 			dev_err(ctx->dev, "unable to load firmware\n");
 			return ret;
-- 
2.21.0

